hamilton/src/hybrid_planner.cpp

```diff
@@ -0,0 +1,282 @@
+#include "hamilton/hybrid_planner.h"
+
+namespace Hamilton
+{   
+  HybridPlanner::~HybridPlanner()
+  {
+    //)WHAT)TODO. Reset Descartes robot model ptr? What about MoveIt? 
+  }
+
+  void HybridPlanner::init()
+  { 
+    // Load params from YAML file into HybridPlanner::HybridConfiguration config_ 
+    if(node_handle_.getParam("group_name",config_.group_name) &&
+    node_handle_.getParam("robot_description",config_.robot_description) &&
+    node_handle_.getParam("tcp_frame",config_.tcp_frame) &&
+    node_handle_.getParam("base_link",config_.base_link) &&
+    node_handle_.getParam("world_frame",config_.world_frame) &&
+    node_handle_.getParam("controller_joint_names",config_.joint_names) &&
+    node_handle_.getParam("moveit_params/plan_time", config_.plan_time) &&
+    node_handle_.getParam("moveit_params/step_size", config_.step_size) &&
+    node_handle_.getParam("moveit_params/jump_thresh", config_.jump_thresh) &&
+    node_handle_.getParam("moveit_params/replan", config_.moveit_replan) &&
+    node_handle_.getParam("moveit_params/avoid_collision", config_.avoid_collision) &&
+    node_handle_.getParam("moveit_params/moveit_planner_type",config_.moveit_planner_type) &&
+    node_handle_.getParam("descartes_params/descartes_planner_type",config_.descartes_planner_type) &&
+    node_handle_.getParam("trajectory/time_delay",config_.time_delay) &&
+    node_handle_.getParam("trajectory/time_offset",config_.time_offset))
+    {
+      ROS_INFO_STREAM("Loaded parameters");
+    }
+    else
+    {
+      ROS_ERROR_STREAM("Failed to load application parameters");
+      exit(-1); //TO CHECK
+    }
+
+    // Set the group name of HybridPlanner's MoveIt! group member in accordance to its config_ struct
+    moveit::planning_interface::MoveGroup local_group(config_.robot_description);
+    moveit_group_ = local_group;
+    
+    // Initialize a Descartes robot model, MoveIt group is already initialized to config_.group_name in HybridPlanner class // TOCHECK here v/s in class definition itself
+    if (!descartes_model_->initialize(config_.robot_description, config_.group_name, config_.world_frame, config_.tcp_frame))
+    {
+      ROS_WARN("Could not initialize Descartes robot model");
```

Comment:
Agreed. But if only free paths segments are planned?
It was `ROS_ERROR_STREAM` earlier, in my local repo. Maybe, error is better only. Checl #L60